kani-compiler/src/kani_middle/transform/kani_intrinsics.rs

```diff
@@ -120,11 +127,141 @@ impl IntrinsicGeneratorPass {
+                            get_kani_sm_function(tcx, "KaniMemInitSMGet"),
+                            &GenericArgs(vec![
+                                GenericArgKind::Const(
+                                    TyConst::try_from_target_usize(
+                                        layout.to_byte_mask().len() as u64
+                                    )
+                                    .unwrap(),
+                                ),
+                                GenericArgKind::Type(*pointee_info.ty()),
+                            ]),
+                        )
+                        .unwrap();
+                        let layout_operand = mk_layout_operand(
+                            &mut new_body,
+                            &mut terminator,
+                            InsertPosition::Before,
+                            layout,
+                        );
+                        new_body.add_call(
+                            &shadow_memory_get_instance,
+                            &mut terminator,
+                            InsertPosition::Before,
+                            vec![
+                                Operand::Copy(Place::from(1)),
+                                layout_operand,
+                                Operand::Copy(Place::from(2)),
+                            ],
+                            Place::from(ret_var),
+                        );
+                    }
+                    PointeeLayout::Slice { element_layout } => {
+                        let shadow_memory_get_instance = Instance::resolve(
+                            get_kani_sm_function(tcx, "KaniMemInitSMGetSlice"),
+                            &GenericArgs(vec![
+                                GenericArgKind::Const(
+                                    TyConst::try_from_target_usize(
+                                        element_layout.to_byte_mask().len() as u64,
+                                    )
+                                    .unwrap(),
+                                ),
+                                GenericArgKind::Type(*pointee_info.ty()),
+                            ]),
+                        )
+                        .unwrap();
+                        let layout_operand = mk_layout_operand(
+                            &mut new_body,
+                            &mut terminator,
+                            InsertPosition::Before,
+                            element_layout,
+                        );
+                        new_body.add_call(
+                            &shadow_memory_get_instance,
+                            &mut terminator,
+                            InsertPosition::Before,
+                            vec![Operand::Copy(Place::from(1)), layout_operand],
+                            Place::from(ret_var),
+                        );
+                    }
+                    PointeeLayout::TraitObject => unimplemented!(),
```

Comment:
can you add an unsupported check like the one bellow?

Reply:
I think I meant to just leave that case empty, sorry. We generate a failing assertion for any unsize cast right now, so this case presumably won't even happen.

Reply:
If that's the case, please replace this by `unreachable!()`.

Reply:
Talked offline. Added an `assert(false)` for cases where a pointer to trait objects is used